chapters/generative-models.md

```diff
@@ -355,23 +369,29 @@ In the above example we take three steps to compute the output value: we sample
 To make this more clear let us re-write the program as:
 
 ~~~~
-var A = flip();
-var B = flip();
-var C = [A, B];
-C;
+var a = flip()
+var b = flip()
+var c = [a, b]
+c
 ~~~~
 
 We can directly observe (as we did above) that the probability of `true` for `A` is 0.5, and the probability of `false` from `B` is 0.5. Can we use these two probabilities to arrive at the probability of 0.25 for the overall outcome `C` = `[true, false]`? Yes, using the *product rule* of probabilities:
 The probability of two random choices is the product of their individual probabilities.
 The probability of several random choices together is often called the *joint probability* and written as $$P(A,B)$$.
 Since the first and second random choices must each have their specified values in order to get `[true, false]` in the example, the joint probability is their product: 0.25.
 
-We must be careful when applying this rule, since the probability of a choice can depend on the probabilities of previous choices. For instance, compute the probability of `[true, false]` resulting from this program:
+We must be careful when applying this rule, since the probability of a choice can depend on the probabilities of previous choices. For instance, we can visualize the the exact probability of `[true, false]` resulting from this program using `Infer` with `enumerate`:
 
 ~~~~
-var A = flip();
-var B = flip(A ? 0.3 : 0.7);
-[A, B];
+var a = flip()
+var b = flip(a ? 0.3 : 0.7)
+
+Infer({method: 'enumerate'}, function () {
```

Comment:
In this chapter I'd prefer to stick with only the 'forward' inference method, since we haven't yet introduce inference as a more general thing. (If we need to let students know about exact computation using 'enumerate' for the HW we can do it there.)